rumqttd/src/protocol/mod.rs

```diff
@@ -596,17 +596,33 @@ pub fn qos(num: u8) -> Option<QoS> {
 }
 
 /// Checks if a topic or topic filter has wildcards
-pub fn has_wildcards(s: &str) -> bool {
-    s.contains('+') || s.contains('#')
+pub fn has_wildcards(s: impl AsRef<str>) -> bool {
+    s.as_ref().contains(['+', '#'])
 }
 
-/// Checks if a topic is valid
-pub fn valid_topic(topic: &str) -> bool {
-    if topic.contains('+') {
-        return false;
-    }
+/// Check if a topic is valid for PUBLISH packet.
+pub fn valid_topic(topic: impl AsRef<str>) -> bool {
+    is_valid_topic_or_filter(&topic) && !has_wildcards(topic)
+}
 
-    if topic.contains('#') {
+pub const MAX_TOPIC_LEN: usize = 65535;
```

Comment:
There's a link to the MQTT standard obove. Can you please add this here as well?